lib/spack/spack/test/cmd/install.py

```diff
@@ -64,6 +80,15 @@ def test_install_package_and_dependency(
     assert not spack.repo.get(s).stage.created
 
 
+@pytest.mark.usefixtures('mock_calls_for_install', 'builtin_mock', 'config')
+def test_install_runtests():
+    install('--test=root', 'dttop')
+    assert spack.package_prefs.PackageTesting.test.call_count == 1
```

Comment:
Seems like you could check not just the call count but whether the only thing in there is `dttop`.   `PackageTesting` has the information since `test()` is passed the package name, right?  Checking for 1 call is necessary but not sufficient.